executor/instruction/relates_reverse_executor.rs

```diff
@@ -112,8 +112,7 @@ impl RelatesReverseExecutor {
                 let relates: Vec<_> = self
                     .role_relation_types
                     .keys()
-                    .map(|role| role.as_role_type().get_relates(&**snapshot, type_manager))
-                    .map_ok(|set| set.to_owned())
+                    .map(|role| role.as_role_type().get_relates_root(&**snapshot, type_manager))
```

Comment:
Hmm are we sure about this one? It's not making sense to me - pull me in to explain please

Reply:
You might be right, maybe I need to collect all the relates here, not only the root ones. Let's discuss how this thing is used

Reply:
This change doesn't affect the old logic, so we merge it this way, but we'll revisit the logic later